### PR TITLE
Add support for .dockerignore specified by non-default Dockerfile name

### LINKED ISSUE #3231: Add support for Dockerfile.dockerignore files
From the tilt channel:
> Looks like tilt doesn’t support docker_build from Dockerfile.foo with related Dockerfile.foo.dockerignore. Can anybody confirm this ?

I haven't been able to find official specification of this feature, only a brief reference to it in release notes and in random github comments
https://github.com/moby/moby/issues/12886#issuecomment-480575928
and here
https://github.com/docker/engine/pull/215

It's supposed to be implemented on the buildkit side, though we need to dig in a bit about why it's not working out of the box

### PR DESCRIPTION
**Pull request by salim-runsafe**
Fixes #3231 

I'm not super confident in how this impacts docker-compose, but all tests pass.

### CODE REVIEW

File: internal/tiltfile/docker_compose.go [hunk 1]
```diff
@@ -301,7 +301,7 @@ func (s *tiltfileState) dcServiceToManifest(service *dcService, dcSet dcResource
 	paths = append(paths, dcInfo.LocalPaths()...)
 	paths = append(paths, filepath.Dir(dcSet.tiltfilePath))
 
-	dcInfo = dcInfo.WithDockerignores(s.dockerignoresFromPathsAndContextFilters(paths, []string{}, []string{}))
```
Comment (nicks) [inline internal/tiltfile/docker_compose.go:304]: i think you need to pass `service.DfPath` here

Comment (salim-runsafe) [reply internal/tiltfile/docker_compose.go:304]: That's exactly what I've been trying to figure out for the last half hour. I wish I'd seen your comment sooner!

File: internal/tiltfile/tiltfile_docker_compose_test.go [hunk 2]
```diff
@@ -157,6 +157,39 @@ services:
...
+	f.load("baz")
+	configPath := f.TempDirFixture.JoinPath("docker-compose.yml")
+	f.assertDcManifest("baz",
+		dcConfigPath([]string{configPath}),
+		dcYAMLRaw(dcYAML),
+		dcDfRaw(simpleDockerfile),
+		// TODO(maia): assert m.tiltFilename
+	)
+
+	expectedConfFiles := []string{"Tiltfile", ".tiltignore", "docker-compose.yml", "baz/alternate-Dockerfile", "baz/alternate-Dockerfile.dockerignore"}
```
Comment (salim-runsafe) [inline internal/tiltfile/tiltfile_docker_compose_test.go:188]: @nicks can you confirm that this should not expect `.dockerignore` in the root directory?

Comment (nicks) [reply internal/tiltfile/tiltfile_docker_compose_test.go:188]: yep!

**Comment by salim-runsafe** [comment]
Thanks for your help with this @nicks. Do you know when the release will be that incorporates this change?

**Comment by nicks** [comment]
i suspect we'll do a release today or tomorrow

**Comment by nicks** [comment]
(but no promises, depends on some other vetting and requests)

**Pull request merged**